pkg/ibmcsidriver/controller.go

```diff
@@ -437,6 +439,24 @@ func (csiCS *CSIControllerServer) CreateSnapshot(ctx context.Context, req *csi.C
 	ctxLogger.Info("CSIControllerServer-CreateSnapshot... ", zap.Reflect("Request", *req))
 	defer metrics.UpdateDurationFromStart(ctxLogger, "CreateSnapshot", time.Now())
 
+	//Feature flag to avoid default enablement of CreateSnapshot feature.
+	if  strings.ToLower(os.Getenv("IS_SNAPSHOT_ENABLED")) != "true" {
+		ctxLogger.Info("CreateSnapshot functionality is disabled.")
+		ctxLogger.Info("CSIControllerServer-CreateSnapshot... is not enabled. Calls to VPC IAAS Snapshot Service are disabled")
+		time.Sleep(10 * time.Second) //To avoid multiple retries from kubernetes to CSI Driver
+		return nil, commonError.GetCSIError(ctxLogger, commonError.MethodUnimplemented, requestID, nil, "CreateSnapshot functionality is disabled.")
+	}
+
+	maxDelaySnapshotCreate := MAX_DELAY_SNAPSHOT_CREATE // 300 seconds default
```

Comment:
can we add this just before calling createSnapshot at line number 500, may be create new method for this